ioeffect/src/main/scala/scalaz/ioeffect/RTS.scala

```diff
@@ -73,6 +73,12 @@ trait RTS {
     result.get
   }
 
+  final def unsafeShutdownAndWait(timeout: Duration): Unit = {
+    threadPool.shutdown()
+    threadPool.awaitTermination(timeout.toMillis, TimeUnit.MILLISECONDS)
```

Comment:
shouldn't `scheduledExecutor` be shutdown as well?

Reply:
Maybe. Originally I didn't include it coz it's lazy. But it could be included in the shutdown.
WDYT @jdegoes ?

Reply:
Without it this:

```
object IOTest4 extends App with RTS {
  unsafePerformIO(IO.sleep[Void](1.second).flatMap(_ => IO.sync(println("Done"))).fork.flatMap(_.join))
  threadPool.shutdown()
  //scheduledExecutor.shutdown()
  println("Exit")
}
```

never ends

Reply:
We have to terminate it.

Reply:
I'm calling `sys.exit` after the shutdown, so it will exit. The only case it would wait for non-daemon threads is `ExitStatus.DoNotExit`

Reply:
Check out the branch and give it a go, I've tested all the scenarios and it worked so far.

Reply:
This change should be made in `scalaz/scalaz`, btw, and then backported here, just so we have one definitive reference (until `scalaz-ioz` becomes the default—hopefully this week!).

Reply:
> If you want to exit immediately, return ExitNow. If you want to wait for jobs to finish before exiting, return ExitWhenDone. And if you want to keep the JVM alive as long as non-daemon threads exist, return DoNotExit.

👍

Reply:
@jdegoes I think it is a backport
@ktonga ah yes, sys.exit solves the problem of any remaining threads definitively ;)

Reply:
Ok, will raise a PR on z8 before this one gets merged

Reply:
> until scalaz-ioz becomes the default—hopefully this week!

Working on it!